.github/workflows/pre-release.yml

```diff
@@ -95,47 +83,75 @@ jobs:
     - name: Set up Docker Buildx
       id: buildx
-      uses: docker/setup-buildx-action@v1
+      uses: docker/setup-buildx-action@v2
       with:
         driver-opts: network=host
     - name: Build ${{ matrix.edition }} container
-      uses: docker/build-push-action@v2
+      uses: docker/build-push-action@v3
       with:
         context: bin/docker/.
         platforms: linux/amd64
         network: host
-        tags: localhost:5000/local-metabase:${{ env.IMAGE_TAG }}
+        tags: localhost:5000/local-metabase:${{ github.event.inputs.commit }}
         no-cache: true
         push: true
 
     - name: Launch container
-      run: docker run --rm -dp 3000:3000 localhost:5000/local-metabase:${{ env.IMAGE_TAG }}
+      run: docker run --rm -dp 3000:3000 localhost:5000/local-metabase:${{ github.event.inputs.commit }}
       timeout-minutes: 5
     - name: Wait for Metabase to start
       run: while ! curl -s 'http://localhost:3000/api/health' | grep '{"status":"ok"}'; do sleep 1; done
       timeout-minutes: 3
 
+    - name: Login to Docker Hub
+      uses: docker/login-action@v2
+      with:
+        username: ${{ secrets.DOCKERHUB_RELEASE_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_RELEASE_TOKEN }}
     - name: Determine the target Docker Hub repository
       run: |
-        if [[ ${{ github.repository }} == 'metabase/metabase' ]]; then
-          if [[ ${{ matrix.edition }} == ee ]]; then
-            echo "Metabase EE: image ${{ env.IMAGE_TAG }} is going to be pushed to metabase/test-metabase-enterprise"
-            echo "DOCKERHUB_REPO=metabase/test-metabase-enterprise" >> $GITHUB_ENV
-          else
-            echo "Metabase OSS: image ${{ env.IMAGE_TAG }} is going to be pushed to metabase/test-metabase"
-            echo "DOCKERHUB_REPO=metabase/test-metabase" >> $GITHUB_ENV
-          fi
+        if [[ ${{ matrix.edition }} == ee ]]; then
+          echo "Metabase EE: image is going to be pushed to ${{ github.repository_owner }}/metabase-enterprise:${{ github.event.inputs.commit }}"
+          echo "DOCKERHUB_REPO=${{ github.repository_owner }}/metabase-enterprise:${{ github.event.inputs.commit }}" >> $GITHUB_ENV
         else
-          echo "DOCKERHUB_REPO=${{ github.repository_owner }}/${{ secrets.CUSTOM_RELEASE_REPO }}" >> $GITHUB_ENV
+          echo "Metabase OSS: image is going to be pushed to ${{ github.repository_owner }}/metabase:${{ github.event.inputs.commit }}"
+          echo "DOCKERHUB_REPO=${{ github.repository_owner }}/metabase:${{ github.event.inputs.commit }}" >> $GITHUB_ENV
         fi
-    - name: Login to Docker Hub
-      uses: docker/login-action@v1
+    - name: Retag and push container image to Docker Hub
+      run: |
+        echo "Pushing container image ${{ env.DOCKERHUB_REPO }} ..."
+        docker tag localhost:5000/local-metabase:${{ github.event.inputs.commit }} ${{ env.DOCKERHUB_REPO }}
+        docker push ${{ env.DOCKERHUB_REPO }}
+        echo "Finished!"
+
+  verify-docker-pull:
```

Comment:
WDYT if we have `verify-docker-pull` only as a **step** in a previous job, rather than having to spin up new container for a completely new **job**?

That way we can save some resources and we can re-use the conditional logic re: `DOCKERHUB_REPO`. That could dry the code up significantly.

Reply:
If we do that, the `docker pull` will only pull from its local cache (since the container image was constructed a few seconds) and not from Dockerhub.
The reason for a separate job is so that the `push` and `pull` are isolated (they don't share the execution environment).

Reply:
And environment variables are not persisted between jobs, right?

Reply:
The env variables can be scoped to the entire workflow only if they are defined statically at the top level.

TBH I personally wouldn't worry much about duplication for now, since at some point it can be abstracted away (the case with our custom actions e.g. `prepare-frontend` etc).